### ISSUE TITLE
adam-submit transform fails with Exception in thread "main" java.lang.IncompatibleClassChangeError: Implementing class 

### ISSUE THREAD

**Issue by PrakashShanmugam** [bigdatagenomics/adam#1306]
The adam-submit command fails with the following exception

Exception in thread "main" java.lang.IncompatibleClassChangeError: Implementing class
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)

This was tested in macOS sierra machine with java 1.8.0_111 with spark 2.0.

The github instruction was followed and made clean package with the same machine. It fails at this step
adam-submit transform $ADAM_HOME/adam-core/src/test/resources/small.sam /tmp/small.adam

**Comment by heuermh** [comment]
Hello @PrakashShanmugam!

To confirm, the steps you took were something like
```bash
$ git clone https://github.com/bigdatagenomics/adam.git
$ cd adam
$ mvn clean package
$ export $ADAM_HOME=`pwd`
$ export $PATH=$ADAM_HOME/bin:$PATH
$ adam-submit transform $ADAM_HOME/adam-core/src/test/resources/small.sam /tmp/small.adam
```
And then when you run adam-submit with Spark 2.0 installed you get the error above.

I believe this is because building from source builds against Spark 1.6.3 by default, and those artifacts won't work when run with Spark 2.0.x.

If you run the script to reconfigure the build against Spark 2.0.x and then clean package
```bash
$ ./scripts/move_to_spark_2.sh
$ mvn clean package
```
you should be able to run with Spark 2.0.x.  We should update the README to mention this.

**Comment by PrakashShanmugam** [comment]
Hi,

Yes. Now, it works fine.

Thank you very much.


Regards,

Prakash Shanmugam.

**Comment by heuermh** [comment]
Great, thank you!  We'll get the README fixes in for #1307 soon.
